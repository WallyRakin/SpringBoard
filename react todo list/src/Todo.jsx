// Todo.js
import React from 'react';

export function Todo({ id, task, removeTodo }) {
    return (
        <div>
            {task}
            <button onClick={() => removeTodo(id)}>X</button>
        </div>
    );
};