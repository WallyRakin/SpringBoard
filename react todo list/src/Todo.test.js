import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Todo } from './Todo';

it('Todo renders without crashing', () => {
    render(<Todo id={1} task="Test Task" removeTodo={() => { }} />);
});

it('Todo snapshot', () => {
    const tree = renderer.create(<Todo id={1} task="Test Task" removeTodo={() => { }} />).toJSON();
    expect(tree).toMatchSnapshot();
});
