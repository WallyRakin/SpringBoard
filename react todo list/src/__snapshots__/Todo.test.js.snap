// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo snapshot 1`] = `
<div>
  Test Task
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
